.github/workflows/sync-translations-poeditor.yml

```diff
@@ -0,0 +1,185 @@
+        run: |
+          branch_name="update-translations-messages"
+
+          # Check if branch exists on remote
+          if git ls-remote --exit-code origin $branch_name; then
+            echo "Branch $branch_name already exists. Fetching it..."
+            git fetch origin $branch_name:$branch_name
+
+            git checkout $branch_name
+            git pull origin $branch_name
+          else
+            echo "Creating new branch $branch_name..."
+            git checkout -b $branch_name
+          fi
+
+      - name: Extract i18n keys
+        run: npm run extract-i18n
+
+      - name: Upload translations to POEditor
+        run: |
+          declare -A files
+
+          # Populate file mappings dynamically
+          files["en"]="src/assets/locale/messages.xlf"
+          files["de"]="src/assets/locale/messages.de.xlf"
+          files["it"]="src/assets/locale/messages.it.xlf"
+          files["fr"]="src/assets/locale/messages.fr.xlf"
+
+          for lang in "${!files[@]}"; do
+            echo "Uploading ${files[$lang]} for language $lang"
+            response=$(curl -s -X POST https://api.poeditor.com/v2/projects/upload \
+              -F "api_token=${{ secrets.POEDITOR_API_TOKEN }}" \
+              -F "id=${{ secrets.POEDITOR_PROJECT_ID }}" \
+              -F "language=$lang" \
+              -F "updating=terms_translations" \
+              -F "overwrite=1" \
+              -F "sync_terms=1" \
+              -F "file=@${files[$lang]}")
+            status=$(echo $response | jq -r '.response.status')
+            if [ "$status" != "success" ]; then
+              echo "Error uploading file for $lang: $response"
+              exit 1
+            fi
+            echo "Waiting for 30 second to respect rate limits..."
+            sleep 30
+          done
+
+      - name: Export translations from POEditor
+        run: |
+          languages=("en" "fr" "it" "de")
+          for lang in "${languages[@]}"; do
+            echo "Exporting translations for language $lang"
+            response=$(curl -s -X POST https://api.poeditor.com/v2/projects/export \
+              -F "api_token=${{ secrets.POEDITOR_API_TOKEN }}" \
+              -F "id=${{ secrets.POEDITOR_PROJECT_ID }}" \
+              -F "language=$lang" \
+              -F "type=xlf")
+            status=$(echo $response | jq -r '.response.status')
+            if [ "$status" == "success" ]; then
+              url=$(echo $response | jq -r '.result.url')
+              if [ -n "$url" ]; then
+                echo "Downloading translations for $lang from $url"
+                if [ "$lang" == "en" ]; then
+                  curl -s -o src/assets/locale/messages.xlf $url
+                else
+                  curl -s -o src/assets/locale/messages.$lang.xlf $url
+                fi
+              else
+                echo "No URL found for downloading translations for $lang"
+                exit 1
+              fi
```

Comment:
This is the POEditor export API, where we can export the updated language files. For reference, here is the documentation of the [POEditor Export API](https://poeditor.com/docs/api#projects_export).